xarray/core/groupby.py

```diff
@@ -66,6 +67,52 @@ def _dummy_copy(xarray_obj):
         raise AssertionError
     return res
 
+def _is_one_or_none(obj):
+    return obj == 1 or obj is None
```

Comment:
do you want to treat `True` as `1`?  If not, this should be `obj is 1 or obj is None`.